File: apps/api/AssetManagement.Api/DTOs/NotificationDtos.cs
namespace AssetManagement.Api.DTOs;

public record NotificationItem(string Id, string Name, DateTime ExpiryDate);

public record NotificationGroup(int Count, List<NotificationItem> Items);

public record NotificationSummary(
    int TotalCount,
    NotificationGroup Warranties,
    NotificationGroup Certificates,
    NotificationGroup Licences);

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/dto/NotificationDtos.kt
package com.assetmanagement.api.dto

import java.time.Instant
import java.util.*

data class NotificationItem(
    val id: UUID,
    val name: String,
    val expiryDate: Instant
)

data class NotificationGroup(
    val count: Int,
    val items: List<NotificationItem>
)

data class NotificationSummary(
    val totalCount: Int,
    val warranties: NotificationGroup,
    val certificates: NotificationGroup,
    val licences: NotificationGroup
)
